src/parser.cpp

```diff
@@ -1081,7 +1077,7 @@ namespace ada::parser {
           // Otherwise, run these steps:
           else {
             // UTF-8 percent-encode c using the path percent-encode set and append the result to buffer.
-            buffer += unicode::utf8_percent_encode(std::string{*pointer}, character_sets::PATH_PERCENT_ENCODE);
+            buffer.append(unicode::utf8_percent_encode(std::string{*pointer}, character_sets::PATH_PERCENT_ENCODE));
```

Comment:
So here, buffer.append might throw (in theory), if it does and you have declare the function noexcept, you may get a hard crash.

Reply:
If I'm not mistaken, it only throws in the following case, [referencing docs](https://cplusplus.com/reference/string/string/append/), which is not the case for us.

```
If this is greater than str's [length](https://cplusplus.com/string::length), it throws [out_of_range](https://cplusplus.com/out_of_range).
```